lib/Core/SQLiteBuildDB.cpp

```diff
@@ -640,14 +581,27 @@ class SQLiteBuildDB : public BuildDB {
 
   
 private:
+  /// Local cache of database DBKeyID (values) to engine KeyIDs
+  llvm::DenseMap<uint64_t, KeyID> engineKeyIDs;
```

Comment:
FWIW, DenseMap should offer some kind of traits you can implement to allow this to be a DBKeyID -> KeyID map.